stream_alert/alert_processor/outputs/carbonblack.py

```diff
@@ -0,0 +1,108 @@
+"""
+Copyright 2017-present, Airbnb Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+from collections import OrderedDict
+from cbapi.response import BannedHash, Binary, CbResponseAPI
+from stream_alert.alert_processor import LOGGER
+
+from stream_alert.alert_processor.outputs.output_base import (
+    OutputDispatcher,
+    OutputProperty,
+    StreamAlertOutput
+)
+
+
+@StreamAlertOutput
+class CarbonBlackOutput(OutputDispatcher):
+    """CarbonBlackOutput handles all alert dispatching for CarbonBlack"""
+    __service__ = 'carbonblack'
+
+    @classmethod
+    def get_user_defined_properties(cls):
+        """Get properties that must be asssigned by the user when configuring a new CarbonBlack
+        output.  This should be sensitive or unique information for this use-case that needs
+        to come from the user.
+
+        Returns:
+          OrderedDict: Contains various OutputProperty items
+        """
+        return OrderedDict([
+            ('descriptor',
+             OutputProperty(description='a short and unique descriptor for this'
+                                        ' carbonblack output')),
+            ('url',
+             OutputProperty(description='URL to the CB Response server [https://hostname]:',
+                            mask_input=True,
+                            cred_requirement=True)),
+            ('ssl_verify',
```

Comment:
this could actually be problematic, since these inputs are always `str` types.. basically, when this prompts the user they'll enter something like `true`. this will get serialized to a string and saved to s3 like:
```json
{
    "url": "whatever.url",
    "ssl_verify": "true",
    "token": "your.token"
}
```

When this is passed to the API below ([here](https://github.com/airbnb/streamalert/pull/713/files#diff-57f77691016d4f3f105484b13ad81280R77) it should be `bool` == `True` or `False` and instead will be a string.

I'm actually not sure of the best way to handle this, since these input don't have any validation right now. The StreamAlert apps have a 'format' param that can be a validation function that I would like to adopt here (eventually). An example can be seen here: https://github.com/airbnb/streamalert/blob/master/app_integrations/apps/box.py#L190-L210

Reply:
Are we looking to make this change now or later?

Reply:
that's a good questions - what are the odds that anyone would ever want ssl_verify to be `False`? my short term fix would be to make the ssl_verify attribute hardcoded to the boolean of `True`. thoughts?

Reply:
For now I think this would be the ideal situation.  We could fix later when we come up with a better solution.  The `ssl_verify` flag is also optional.